x/gov/client/cli/tx.go

```diff
@@ -82,22 +83,35 @@ $ gaiacli gov submit-proposal --title="Test Proposal" --description="My awesome
 				WithCodec(cdc).
 				WithAccountDecoder(cdc)
 
-			fromAddr, err := cliCtx.GetFromAddress()
+			// Get from address
+			from, err := cliCtx.GetFromAddress()
 			if err != nil {
 				return err
 			}
 
+			// Pull associated account
+			account, err := cliCtx.GetAccount(from)
+			if err != nil {
+				return err
+			}
+
+			// Find deposit amount
 			amount, err := sdk.ParseCoins(proposal.Deposit)
 			if err != nil {
 				return err
 			}
 
+			// ensure account has enough coins
```

Comment:
This isn't wrong, but it's a weird way to write this logic - we shouldn't be doing this in CLI commands generally, we should be simulating transactions and displaying the error to the user before broadcasting them (if the simulation failed).

If we simulate a submit-proposal transaction with not enough sender balance, it should fail.

Reply:
Ok. I copied this pattern from the `x/auth/client/cli/sign.go`. Is there a place where this pattern is shown? The code in this PR fixes some really bad errors and UX in the governance and I works well in the current form.

Reply:
Discussed, fine for now, let's fix the root cause though (separately) - https://github.com/cosmos/cosmos-sdk/issues/2953